### PR TITLE
[SPARK-43839][SQL] Convert `_LEGACY_ERROR_TEMP_1337` to `UNSUPPORTED_FEATURE.TIME_TRAVEL`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to convert `_LEGACY_ERROR_TEMP_1337` to `UNSUPPORTED_FEATURE.TIME_TRAVEL` and remove `_LEGACY_ERROR_TEMP_1335`

### Why are the changes needed?
- The changes improve the error framework.
- In the spark base code `_ LEGACY_ ERROR_ TEMP_ 1335` is no longer used anywhere.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Add new UT
- Pass GA

### CODE REVIEW

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala:92]: can we remove `tableNotSupportTimeTravelError` from `QueryCompilationErrors`? @panbingkun @beliefer

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala:89]: do we still need the `if-else`?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala
Comment (panbingkun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala:92]: Let me investigate it.

Comment (panbingkun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala:92]: Its corresponding error class '_LEGACY_ERROR_TEMP_1337' has been merged with `UNSUPPORTED_FEATURE.TIME_TRAVEL`, so it has been removed here.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala
Comment (panbingkun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala:89]: Yeah, after merging, the code here can be simplified. It's my bad, let me do it immediately.

Comment (panbingkun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala:89]: @cloud-fan 
Follow Up pr: https://github.com/apache/spark/pull/42082

**Pull request closed without merging**
